controllers/space/space_controller.go

```diff
@@ -47,10 +47,10 @@ type Reconciler struct {
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, memberClusters map[string]cluster.Cluster) error {
 	b := ctrl.NewControllerManagedBy(mgr).
 		// watch Spaces in the host cluster
-		For(&toolchainv1alpha1.Space{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		For(&toolchainv1alpha1.Space{}, builder.WithPredicates(predicate.GenerationChangedPredicate{}, predicate.AnnotationChangedPredicate{})).
 		Watches(&source.Kind{Type: &toolchainv1alpha1.NSTemplateTier{}},
 			handler.EnqueueRequestsFromMapFunc(MapNSTemplateTierToSpaces(r.Namespace, r.Client)),
-			builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+			builder.WithPredicates(predicate.GenerationChangedPredicate{}, predicate.AnnotationChangedPredicate{})).
```

Comment:
why do you need to change the predicate for NSTemplateTier CRD?

Reply:
In case the annotation is deleted (or added) by accident on the NSTemplateSet side. Then the space controller would revert it. It should not happen because we don't actually usually touch NSTemplateSets manually but I guess that wouldn't hurt.

Reply:
but you modified the watch of the NSTemplateTier, not NSTemplateSet. 
We do watch literally all changes in NSTemplateSet CRs (we don't have any predicate there) - see https://github.com/alexeykazakov/host-operator/blob/d8fa9d5139bb2294a840f089e2a37bb4c244a77d/controllers/space/space_controller.go#L57-L61

Reply:
Oh... yes, it's `NSTemplateTier` indeed! Let me revert this. Good catch! https://github.com/codeready-toolchain/host-operator/pull/1063